src/transformers/models/whisper/tokenization_whisper.py

```diff
@@ -705,3 +705,272 @@ def get_decoder_prompt_ids(self, task=None, language=None, no_timestamps=True):
         forced_tokens = self.prefix_tokens[1:]
         forced_decoder_ids = [(rank + 1, token) for rank, token in enumerate(forced_tokens)]
         return forced_decoder_ids
+
+    def _decode_asr(self, model_outputs, *, return_timestamps, return_language, time_precision):
+        """
+        Internal method meant to only be used by asr pipeline. Handles all the little quirks specific to whisper to
+        handle the various options not allowed in other seq2seq models
+        """
+
+        # =========== Overview ============
+        # - iterate over all outputs
+        # - all tokens within output
+        # - Each token can be
+        #   - language token
+        #   - special token
+        #   - timestamp token
+        #   - text token
+        # - We accumulate the text tokens.
+        # - We split on end timestamps
+        # - Lots of complexity comes from stride and timestamps
```

Comment:
Might be good to add comments in the code below matching this summary.

Reply:
Some of it can be aligned, except the *complexity* one, the `each token` which unfortunately is a bit spread out (stop conditions depend on many things to be as early as possible, and still handle conflicts)

Th rest should be aligned.